roles/cis/tasks/4/verify/4.1.18.yml

```diff
@@ -0,0 +1,15 @@
+---
+- name: "4.1.18 Verify that kernel module loading and unloading is collected (Scored)"
+  vars:
+    audit_rules: /etc/audit/rules.d/audit.rules
+  shell: |
+      grep -q '^-w /sbin/insmod -p x -k modules$' "{{ audit_rules }}" && \
```

Comment:
For all simple "grep" commands (for complex grep commands, we can stick to shell), can we use lineinfile module:
Example:
- name: "Ensure /etc/audit/rules.d/audit.rules '^-w /sbin/insmod -p x -k modules$' "{{ audit_rules }}'"
  lineinfile:
    name: /etc/audit/rules.d/audit.rules
    line: "^-w /sbin/insmod -p x -k modules$' "{{ audit_rules }}"
    state: present
  check_mode: yes
  register: result
  failed_when: (result is changed) or (result is failed)

Reply:
For example related to grep, please refer:
https://github.com/vmware/ansible-security-hardening/commit/7a97c83a6f77859e53c086fbe651591cf3ddcb16

roles/cis/tasks/3/verify/3.5.4.yml

Reply:
Thanks for you comment. Since i had not used Blocks, I did not employ the "fail:" construct for simplicity. I will recode the affected files, meanwhile, the current versions can be committed and the recoded one can go as an update.

Reply:
let's please fix now. i would like the current code to fix for @satya-rajesh's comments before check in

Reply:
Addressed all the comments.